packages/server/src/externalTypes/context.ts

```diff
@@ -0,0 +1,4 @@
+export type BaseContext = {};
+
+export type ContextFunction<TContext extends BaseContext = BaseContext> =
```

Comment:
I think this is more like ContextThunk but we should also have a two-generic-arg ContextFunction